tests/containers_test.toit

```diff
@@ -17,3 +28,35 @@ main:
 
   writer := containers.ContainerImageWriter 4096
   writer.close
+
+test_start:
+  sub1 := containers.start containers.current {:}
+  expect_equals 0 sub1.wait
+
+  lambda2_value := null
+  sub2 := containers.start containers.current {:}
+  sub2.on_stopped:: lambda2_value = it
+  expect_equals 0 sub2.wait
+  expect_equals 0 lambda2_value
+
+  lambda3_value := null
+  sub3 := containers.start containers.current {:}
+  expect_equals 0 sub3.wait
+  sub3.on_stopped:: lambda3_value = it
+  expect_equals 0 lambda3_value
+
+  lambda4_called := false
+  lambda4_value := null
+  sub4 := containers.start containers.current {:}
+  sub4.on_stopped::
+    expect_not lambda4_called
+    lambda4_value = it
+    lambda4_called = true
+  // Make sure we get the lambda called before we call
+  // wait on the container. Shouldn't take too long.
+  with_timeout --ms=5_000: while not lambda4_called: sleep --ms=50
+  expect_equals 0 lambda4_value
+  expect_equals 0 sub4.wait
```

Comment:
Add a test for when the container throws an exception?

Reply:
Will do in another PR. It is a little bit tricky, because we seem to sometimes shut down everything when things fail.

Reply:
It's because the primary application container image is automatically marked RUN_BOOT | RUN_CRITICAL. The child container inherits that.